pkg/controller/controller_instance.go

```diff
@@ -1588,19 +1652,40 @@ func (c *controller) processServiceInstanceOperationError(instance *v1beta1.Serv
 // processProvisionSuccess handles the logging and updating of a
 // ServiceInstance that has successfully been provisioned at the broker.
 func (c *controller) processProvisionSuccess(instance *v1beta1.ServiceInstance, dashboardURL *string) error {
-	setServiceInstanceDashboardURL(instance, dashboardURL)
-	setServiceInstanceCondition(instance, v1beta1.ServiceInstanceConditionReady, v1beta1.ConditionTrue, successProvisionReason, successProvisionMessage)
-	instance.Status.ExternalProperties = instance.Status.InProgressProperties
-	clearServiceInstanceCurrentOperation(instance)
-	instance.Status.ProvisionStatus = v1beta1.ServiceInstanceProvisionStatusProvisioned
-	instance.Status.ReconciledGeneration = instance.Status.ObservedGeneration
+	instanceToUpdate := instance // Initially try to update instance directly
 
-	if _, err := c.updateServiceInstanceStatus(instance); err != nil {
-		return err
-	}
+	const interval = 100 * time.Millisecond
+	const timeout = 10 * time.Second
+	err := wait.PollImmediate(interval, timeout, func() (bool, error) {
```

Comment:
As @kibbles-n-bytes described previously, we need this retry loop to correctly handle the situation after we make locks optional, when user updates the spec before controller sends a successful status update.